hikari/impl/event_manager.py

```diff
@@ -371,7 +376,10 @@ async def on_guild_create(  # noqa: C901, CFQ001 - Function too complex and too
             and self._intents & intents_.Intents.GUILD_MEMBERS
             and (payload.get("large") or not presences_declared)
             and (
-                self._cache_enabled_for(config.CacheComponents.MEMBERS)
+                (
+                    self._cache_enabled_for(config.CacheComponents.MEMBERS)
+                    and (self._cache is None or not self._cache.settings.only_my_member)
+                )
```

Comment:
This definitely needs some commenting.

Not only that, maybe it's worth extracting it to a variable above to make it more legible

Reply:
```suggestion
                (
                    # disable auto chunking if we are only interested in the bot member (only_my_member=True)
                    self._cache_enabled_for(config.CacheComponents.MEMBERS)
                    and (self._cache is None or not self._cache.settings.only_my_member)
                )
```